lib/drives/pages/drive_chat_page.dart

```diff
@@ -16,107 +17,118 @@ class DriveChatPage extends StatefulWidget {
 }
 
 class _DriveChatPageState extends State<DriveChatPage> {
-  Drive? _drive;
-  bool _fullyLoaded = false;
+  late Stream<List<Message>> _messagesStream;
 
   @override
   void initState() {
+    List<int> ids = widget.drive.chats!.map((Chat chat) => chat.id!).toList();
+    _messagesStream =
+        SupabaseManager.supabaseClient.from('messages').stream(primaryKey: ['id']).order('created_at').map(
+              (messages) =>
+                  Message.fromJsonList(messages.where((element) => ids.contains(element['chat_id'])).toList()),
+            );
     super.initState();
-    setState(() {
-      _drive = widget.drive;
-    });
-    loadDrive();
+  }
+
+  @override
+  void dispose() {
+    _messagesStream.drain();
+    super.dispose();
   }
 
   @override
   Widget build(BuildContext context) {
-    List<Widget> widgets = [];
-    if (_fullyLoaded) {
-      Set<Profile> riders = _drive!.approvedRides!.map((ride) => ride.rider!).toSet();
-      List<Ride> pendingRides = _drive!.pendingRides!.toList();
-      if (riders.isEmpty && pendingRides.isEmpty) {
-        return Scaffold(
-          appBar: AppBar(
-            title: Text(S.of(context).pageDriveChatTitle),
-          ),
-          body: Center(
-            child: Text(S.of(context).pageDriveChatEmptyMessage),
-          ),
-        );
-      } else {
-        if (riders.isNotEmpty) {
-          List<Widget> riderColumn = [
-            const SizedBox(height: 5.0),
-            Text(
-              S.of(context).pageDriveChatRiderHeadline,
-              style: Theme.of(context).textTheme.titleLarge,
-            ),
-            const SizedBox(height: 10.0),
-            _riderList(riders),
-          ];
-          widgets.addAll(riderColumn);
-        }
-        widgets.add(const SizedBox(height: 10.0));
-      }
-    } else {
-      widgets.add(const SizedBox(height: 10));
-      widgets.add(const Center(child: CircularProgressIndicator()));
-    }
+    List<Chat> chats = widget.drive.chats!.toList();
     return Scaffold(
       appBar: AppBar(
         title: Text(S.of(context).pageDriveChatTitle),
       ),
-      body: RefreshIndicator(
-        onRefresh: loadDrive,
-        child: ListView.separated(
-          itemCount: widgets.length,
-          itemBuilder: (context, index) {
-            return widgets[index];
-          },
-          separatorBuilder: (BuildContext context, int index) {
-            return const SizedBox(height: 10);
-          },
-        ),
-      ),
+      body: chats.isNotEmpty
+          ? StreamBuilder<List<Message>>(
```

Comment:
when going back to this state from the chat_page, I get this error:

Another exception was thrown: Bad state: Stream has already been listened to.

Reply:
for drive_detail or ride_detail? and which chat?

Reply:
can't reproduce that either.